### PR TITLE
Allow literals for GLOBAL IN

### LINKED ISSUE #2610: GLOBAL IN don't return results from all nodes
Why when I use "GLOBAL IN", I do not get results from all nodes?
Example 1:
```
SELECT *
FROM tbl
PREWHERE EntityID GLOBAL IN (        
 SELECT EntityID
 FROM tbl
 PREWHERE EntityID = 'value'
)
```
Example 2:
```
SELECT *
FROM tbl
preWHERE EntityID  = 'value'
```
Count of rows Example 1 less then Example 2.

ClickHouse is deployed on 2 nodes: 1, 2. Each nodes have table with some data and view with aggregated data from all nodes. I connected to node 1 and execute query to view and not see data from node 1. When I connected to node 2 and execute query to view and not see data from node 2.

Why it occurs?

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow literals for GLOBAL IN (i.e. `SELECT * FROM remote('localhost', system.one) WHERE dummy global in (0)`)

Detailed description:
This PR also includes some additional tests:
- missing rows (#2610) 
- distributed over distributed (#9923)
- GLOBAL IN with subquery/function/table/literal

Closes: #2610

### CODE REVIEW

File: src/Interpreters/interpretSubquery.cpp [hunk 1]
```diff
@@ -49,7 +43,7 @@ std::shared_ptr<InterpreterSelectWithUnionQuery> interpretSubquery(
     const auto * table = table_expression->as<ASTIdentifier>();
 
     if (!subquery && !table && !function)
-        throw Exception("Table expression is undefined, Method: ExpressionAnalyzer::interpretSubquery." , ErrorCodes::LOGICAL_ERROR);
+        throw Exception("Table expression is not subquery/table/function.", error_code);
```
Comment (alexey-milovidov) [inline src/Interpreters/interpretSubquery.cpp:46]: I don't remember, why we can see something other than subquery/table/function here?

Comment (azat) [reply src/Interpreters/interpretSubquery.cpp:46]: I guess that because literal is fine for regular `IN`

**Comment by alexey-milovidov** [comment]
But why we are trying to interpret literal as subquery?

**Comment by alexey-milovidov** [comment]
`select * from remote('localhost', system.one) where dummy global in (1);`

Should not it work in the same way is regular IN?

**Comment by azat** [comment]
>Should not it work in the same way is regular IN?

Agree (although that code is used for join, so after the following will be allowed I guess `select 1 union all 2`)

**Comment by alexey-milovidov** [comment]
Let's implement the following logic:

If the right hand side of GLOBAL IN is not a subquery/function/table, either:
- replace GLOBAL IN to IN;
- or do not create external table but pass and then the GLOBAL IN operator will be executed as regular IN (let's check).

> although that code is used for join, so after the following will be allowed I guess select 1 union all 2

Yes, but it is subquery, so it should work fine...

**Comment by azat** [comment]
>replace GLOBAL IN to IN;

Great idea! (I was thinking to do this initially, but by some reason rejected it)

**Comment by azat** [comment]
And I will revert all changes to `interpretSubquery` then

**Comment by azat** [comment]
Description had been update (don't forget to update labels)

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**